File: modules/x-engine-jsi-camera/iOS/Class/JSI_camera.m
//
//  JSI_camera.m
//  camera
//
//  Created by zk on 2020/9/7.
//  Copyright © 2020 edz. All rights reserved.


#import "JSI_camera.h"
#import "JSIContext.h"
#import "XENativeContext.h"
#import "iCamera.h"

@interface JSI_camera()
@property(nonatomic,strong) id<iCamera> camera;
@end

@implementation JSI_camera
JSI_MODULE(JSI_camera)

- (void)afterAllJSIModuleInited {
    self.camera = XENP(iCamera);
}

- (void)_openImagePicker:(_openImagePicker_com_zkty_jsi_camera_0_DTO *)dto complete:(void (^)(NSString *, BOOL))completionHandler {
    CameraParamsDTO *model = [CameraParamsDTO new];
    model.allowsEditing = dto.allowsEditing;
    model.savePhotosAlbum = dto.savePhotosAlbum;
    model.cameraFlashMode = dto.cameraFlashMode;
    model.cameraDevice = dto.cameraDevice;
    model.isbase64 = dto.isbase64;
    model.args = dto.args;
    model.photoCount = dto.photoCount;
    [self.camera openImagePicker:model success:^(NSString *result) {
        completionHandler(result, TRUE);
    }];
}
 
- (void)_saveImageToPhotoAlbum:(_saveImageToPhotoAlbum_com_zkty_jsi_camera_0_DTO *)dto complete:(void (^)(NSString *, BOOL))completionHandler {
    SaveImageDTO *model = [SaveImageDTO new];
    model.type = dto.type;
    model.imageData = dto.imageData;
    [self.camera saveImageToPhotoAlbum:model saveSuccess:^(NSString *result) {
        completionHandler(result, TRUE);
    }];
}
 
 
 
 

@end

File: modules/x-engine-jsi-camera/android/jsi-camera/src/main/java/com/zkty/jsi/JSI_camera.java
package com.zkty.jsi;


import com.zkty.nativ.camera.CameraDTO;
import com.zkty.nativ.camera.ICamera;
import com.zkty.nativ.camera.NativeCamera;
import com.zkty.nativ.camera.SaveCallBack;
import com.zkty.nativ.core.NativeContext;
import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.core.XEngineApplication;
import com.zkty.nativ.jsi.bridge.CompletionHandler;

public class JSI_camera extends xengine_jsi_camera {
    private NativeCamera iCamera;

    @Override
    protected void afterAllJSIModuleInited() {
        NativeModule module = NativeContext.sharedInstance().getModuleByProtocol(ICamera.class);
        if (module instanceof NativeCamera)
            iCamera = (NativeCamera) module;
    }

    @Override
    public void _openImagePicker(_0_com_zkty_jsi_camera_DTO dto, CompletionHandler<String> handler) {
        CameraDTO cameraDTO = new CameraDTO();
        cameraDTO.setAllowsEditing(dto.allowsEditing);
        cameraDTO.setCameraDevice(dto.cameraDevice);
        cameraDTO.setCameraFlashMode(dto.cameraFlashMode);
        cameraDTO.setIsbase64(dto.isbase64);
        cameraDTO.setPhotoCount(dto.photoCount);
        cameraDTO.setSavePhotosAlbum(dto.savePhotosAlbum);
        cameraDTO.setArgs(dto.args);
        XEngineApplication.getCurrentActivity().runOnUiThread(() -> {
            iCamera.openImagePicker(cameraDTO, dto1 -> handler.complete(dto1));
        });

    }


    @Override
    public void _saveImageToPhotoAlbum(_1_com_zkty_jsi_camera_DTO dto, CompletionHandler<String> handler) {
        iCamera.saveImageToAlbum(dto.imageData, dto.type, new SaveCallBack() {
            @Override
            public void saveCallBack() {
                handler.complete("success");
            }
        });
    }
}
